samples: corrected eventTime update.

Arakel2811 opened the pull request:
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-retail/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

Arakel2811 committed:
Corrected eventTime update.

alexander-fenster committed:
Merge branch 'main' into GRST-198-Import-user-events-BQ-update-user-events-json.js

alexander-fenster commented on samples/interactive-tutorials/setup/update-user-events-json.js:
```diff
@@ -23,8 +23,18 @@ async function main() {
+          const yesterday = nowDate - 1000 * 60 * 60 * 24;
+          date = new Date(yesterday);
+        } else {
+          date = nowDate.getDate() - 1;
```
This does not make sense to me. Line 35 will assign an integer number to `date`.

Why do you need to check that `now.getDate() - 1 === 0`? Even today, on August 1st, the old code would work:

```ts
> d = new Date()
2022-08-01T16:57:37.463Z
> d.setDate(d.getDate() - 1)
1659286657463
> d
2022-07-31T16:57:37.463Z
```

Arakel2811 replied:
The previous version of that code didn't work correctly, as it should change eventTime if it is earlier than 90 days, it should change the eventTime to yesterday, but when i changed it with this code now.getDate() - 1, it returned me 0, that's why i added that if statement for check.

Arakel2811 commented:
Hi Alexander,

The previous version of that code didn't work correctly, as it should
change eventTime if it is earlier than 90 days, but when i changed it with
this code now.getDate() - 1, it returned me 0, that's why i added that if
statement for check.

Best regards,
Arakel Galeyan

On Mon, Aug 1, 2022 at 8:59 PM Alexander Fenster ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In samples/interactive-tutorials/setup/update-user-events-json.js
> <https://github.com/googleapis/nodejs-retail/pull/200#discussion_r934737242>
> :
>
> > @@ -23,8 +23,18 @@ async function main() {
>
>      for (let i = 0; i < events.length - 1; ++i) {
>        const event = JSON.parse(`[${events[i]}]`)[0];
> -      const date = new Date(event.eventTime);
> -      date.setDate(date.getDate() - 1);
> +      let date = new Date(event.eventTime);
> +      const nowDate = new Date();
> +      const timeDiff = nowDate - date;
> +
> +      if (timeDiff > 1000 * 60 * 60 * 24 * 90) {
> +        if (nowDate.getDate() - 1 === 0) {
> +          const yesterday = nowDate - 1000 * 60 * 60 * 24;
> +          date = new Date(yesterday);
> +        } else {
> +          date = nowDate.getDate() - 1;
>
> This does not make sense to me. Line 35 will assign an integer number to
> date.
>
> Why do you need to check that now.getDate() - 1 === 0? Even today, on
> August 1st, the old code would work:
>
> > d = new Date()2022-08-01T16:57:37.463Z> d.setDate(d.getDate() - 1)1659286657463> d2022-07-31T16:57:37.463Z
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/googleapis/nodejs-retail/pull/200#pullrequestreview-1057601680>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJF43AXGTQSCISM2YDUF733VW77AFANCNFSM55HTBS7A>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Arakel2811 committed:
Fixed a logical issue of previous commit, did code cleanup.

Arakel2811 committed:
Merge branch 'GRST-198-Import-user-events-BQ-update-user-events-json.js' of github.com:Arakel2811/nodejs-retail into GRST-198-Import-user-events-BQ-update-user-events-json.js

alexander-fenster committed:
Merge branch 'main' into GRST-198-Import-user-events-BQ-update-user-events-json.js
